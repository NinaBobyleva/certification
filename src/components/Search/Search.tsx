import * as S from "./search.styled";

export function Search({setUserName}: {setUserName: React.Dispatch<React.SetStateAction<string>>}) {

    return(
        <S.Input type="text" placeholder="Введите логин пользователя" onChange={(e: React.ChangeEvent<HTMLInputElement>) => setUserName(e.target.value)} />
    );
}