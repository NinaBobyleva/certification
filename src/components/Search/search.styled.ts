import styled from "styled-components";

export const Input = styled.input`
  width: 1100px;
  height: 50px;
  outline: none;
  padding: 20px;
  border-radius: 30px;
  ::placeholder {
    font-weight: 400;
    font-size: 24px;
    line-height: 3px;
    color: #94a6be;
  }
`;
