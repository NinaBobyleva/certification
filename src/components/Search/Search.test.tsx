import renderer from "react-test-renderer";
import { Search } from "./Search";

describe("Компонент Button", () => {
  it("Успешный рендер Button", () => {
    const setLogin = jest.fn();
    const component = renderer
      .create(<Search setLogin={setLogin}/>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});