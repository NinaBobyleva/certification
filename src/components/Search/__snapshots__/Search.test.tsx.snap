// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Button Успешный рендер Button 1`] = `
<input
  className="sc-blHHSb flDCzU"
  onChange={[Function]}
  placeholder="Введите логин пользователя"
  type="text"
/>
`;
